src/main/java/com/zestedesavoir/zestwriter/view/MdConvertController.java

```diff
@@ -98,14 +99,15 @@ public void setMdBox(MdTextController mdBox, Textual extract, Tab tab) throws IO
             BoxEditor.setTop(null);
             BoxRender.setTop(null);
         }
-
+        WordCountLabel.setText("Nombre de mots : " + extract.getMarkdown().replace("'", " ").split(" ").length);
         SourceText.setFont(new Font(config.getEditorFont(), config.getEditorFontsize()));
         SourceText.setStyle("-fx-font-family: \"" + config.getEditorFont() + "\";");
         SourceText.replaceText(extract.getMarkdown());
         SourceText.textProperty().addListener((observableValue, s, s2) -> {
             tab.setText("! " + extract.getTitle());
             this.isSaved = false;
             SourceText.getUndoManager().mark();
+            WordCountLabel.setText("Nombre de mots : " + extract.getMarkdown().replace("'", " ").split(" ").length);
```

Comment:
Ok, je vais faire ça.

Le 4 mai 2016 à 11:29, firm1 <notifications@github.com> a écrit :

> In
> src/main/java/com/zestedesavoir/zestwriter/view/MdConvertController.java
> <https://github.com/firm1/zest-writer/pull/98#discussion_r62011886>:
>
> >          SourceText.setFont(new Font(config.getEditorFont(), config.getEditorFontsize()));
> >          SourceText.setStyle("-fx-font-family: \"" + config.getEditorFont() + "\";");
> >          SourceText.replaceText(extract.getMarkdown());
> >          SourceText.textProperty().addListener((observableValue, s, s2) -> {
> >              tab.setText("! " + extract.getTitle());
> >              this.isSaved = false;
> >              SourceText.getUndoManager().mark();
> > +            WordCountLabel.setText("Nombre de mots : " + extract.getMarkdown().replace("'", " ").split(" ").length);
>
> Étant donné qu'il y'a déjà une classe qui gère les stats sur le texte, tu
> peux l'utiliser gracieusement je pense, il suffit de faire :
>
> Readability readText = new Readability(extract.getMarkdown());WordCountLabel.setText("Nombre de mots : "+readText.getWords())
>
> Tu peux même étendre facilement ta PR en rajoutant d'autres chiffres en
> faisant quelque chose comme :
>
> Readability readText = new Readability(extract.getMarkdown());WordCountLabel.setText("Nombre de mots : "+readText.getWords())SylCountLabel.setText("Nombre de syllabes : "+readText.getSyllables())SentencesCountLabel.setText("Nombre de phrases : "+readText.getSentences())CharCountLabel.setText("Nombre de caractères : "+readText.getCharacters())
>
> par contre un truc qui serait sympa c'est que ce calcul ne soit ps une
> tache bloquante, donc tu peux le lancer à travers un :
>
> Platform.runLater(new Runnable() {
>                 @Override
>                 public void run() {
>                     updateTextInfos()
>                 }
>             });
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/firm1/zest-writer/pull/98/files/07350efcf726081f7b3e4e153dcf807571ff941f#r62011886>
>